### PR TITLE
[det-564] add reset & verify user to python sdk/cli

### PR DESCRIPTION
**Pull request by makk94**
PR bundle:
* https://github.com/preludeorg/backbone/pull/444
* https://github.com/preludeorg/service/pull/1187
* https://github.com/preludeorg/glue/pull/51
* https://github.com/preludeorg/libraries/pull/741

```
(venv) mahinakaholokula@Mahinas-MacBook-Pro my-prelude % prelude --profile personal iam create-account
Overwrite local account credentials for selected profile? [y/N]: y
Enter your email: mahina@preludesecurity.com
(Optional) Enter your name: 
(Optional) Enter your associated company: 
{
  "account": "86c5c33145e3d902f1dc1b1eb45d7af7"
}

Check your email to verify your account.

(venv) mahinakaholokula@Mahinas-MacBook-Pro my-prelude % prelude --profile personal iam account       
{"message":"Unauthorized"}
(venv) mahinakaholokula@Mahinas-MacBook-Pro my-prelude % prelude --profile personal iam verify-user 327bfe0a-1c78-4682-b8c5-56ca5ab5ca36
Overwrite local account credentials for selected profile? [y/N]: y
{
  "account": "86c5c33145e3d902f1dc1b1eb45d7af7",
  "token": "644c5edc-832d-415f-b61a-ab89b025c7a4"
}
(venv) mahinakaholokula@Mahinas-MacBook-Pro my-prelude % prelude --profile personal iam account
{
  "account_id": "86c5c33145e3d902f1dc1b1eb45d7af7",
  "whoami": "mahina@preludesecurity.com",
  "company": "",
  "mode": 0,
  "controls": [],
  "users": [
    {
      "handle": "mahina@preludesecurity.com",
      "permission": 0,
      "name": "",
      "expires": "2024-08-01 12:48:33.582483"
    }
  ],
  "queue": [
    {
      "test": "39de298a-911d-4a3b-aed4-1e8281010a9a",
      "run_code": 1,
      "tag": null,
      "started": "2023-08-02"
    }
  ]
}
(venv) mahinakaholokula@Mahinas-MacBook-Pro my-prelude % prelude --profile personal iam reset-user mahina@preludesecurity.com
{
  "status": true
}

User must check their email to verify their account.

(venv) mahinakaholokula@Mahinas-MacBook-Pro my-prelude % prelude --profile personal iam verify-user ce2670fc-8d06-422c-ac53-88ccc12a065b
Overwrite local account credentials for selected profile? [y/N]: y
{
  "account": "86c5c33145e3d902f1dc1b1eb45d7af7",
  "token": "d5094f4d-22c9-4427-81ba-4899f1db97a8"
}
(venv) mahinakaholokula@Mahinas-MacBook-Pro my-prelude % prelude --profile personal iam verify-user ce2670fc-8d06-422c-ac53-88ccc12a065b
Overwrite local account credentials for selected profile? [y/N]: y
{
  "verified": true
}
```

### CODE REVIEW

File: python/sdk/prelude_sdk/controllers/iam_controller.py [hunk 1]
```diff
@@ -95,6 +95,39 @@ def create_user(self, permission: int, email: str, expires: datetime = None, nam
...
+
+    @verify_credentials
+    def verify_user(self, token: str):
+        """ Verify a user inside an account """
+        res = requests.get(
+            url=f'{self.account.hq}/iam/user',
+            headers=self.account.headers,
+            params=dict(token=token),
+            timeout=10,
+            allow_redirects=False
```
Comment (privateducky) [inline python/sdk/prelude_sdk/controllers/iam_controller.py:119]: why specify allow_redirects=False?

File: python/sdk/prelude_sdk/controllers/iam_controller.py [hunk 2]
```diff
@@ -95,6 +95,39 @@ def create_user(self, permission: int, email: str, expires: datetime = None, nam
...
+        )
+        if res.status_code != 200:
+            raise Exception(res.text)
+
+        res_json = res.json()
+        if 'token' in res_json:
```
Comment (privateducky) [inline python/sdk/prelude_sdk/controllers/iam_controller.py:125]: what's a case that you get a 200 but no token?

**Review by privateducky — CHANGES_REQUESTED** [review]
couple questions

File: python/sdk/prelude_sdk/controllers/iam_controller.py (hunk 1 above)
Comment (makk94) [reply python/sdk/prelude_sdk/controllers/iam_controller.py:119]: ooh. relic of a bygone age when we returned the redirect to platform. i'll remove

Comment (makk94) [reply python/sdk/prelude_sdk/controllers/iam_controller.py:119]: removed

File: python/sdk/prelude_sdk/controllers/iam_controller.py (hunk 2 above)
Comment (makk94) [reply python/sdk/prelude_sdk/controllers/iam_controller.py:124]: anytime you dont input a valid verification_token (either it never existed, or you try to re-use a token that you already successfully verified)

**Review by privateducky — APPROVED** [review]

**Pull request merged**